packages/wagmi/src/client.ts

```diff
@@ -460,7 +462,8 @@ export class Web3Modal extends Web3ModalScaffold {
     chainId,
     connector,
     addresses,
-    status
+    status,
+    resetAccount = true
```

Comment:
Added optional `resetAccount` param - this is something we could discuss. Do you think we should reset the account state when switch to another account? If you see `ChainController.resetAccount`, we are resetting bunch of things and if we call `this.resetAccount` before setting the new account, this is first rendering the unconnected `w3m-button`, then shows account button. Like there is a glitch while switching accounts. I didn't want to have that.

@tomiir @ganchoradkov like to see your thoughts

Reply:
Im wondering if we really need to reset the account. Maybe we should more carefully update values instead

Reply:
I agree with this tbh. There are bunch of values for the account state like the followings. We just need to be double sure that not resetting them wouldn't cause issues on different features. Tried to list them like that. Will make update accordingly:

```js
isConnected: false, // OK - will be updated
smartAccountDeployed: false, // in email/social we don't have multi-account yet, so should be OK to not reset
currentTab: 0, // OK
caipAddress: undefined, // OK - will be updated
address: undefined, // OK - will be updated
balance: undefined, // OK - will be updated
balanceSymbol: undefined, // OK - will be updated
profileName: undefined, // OK - will be updated
profileImage: undefined, // OK - will be updated
addressExplorerUrl: undefined, // OK - will be updated 
tokenBalance: [],  // related with Swaps, need to reset
connectedWalletInfo: undefined, // OK - will be updated 
preferredAccountType: undefined, // in email/social we don't have multi-account yet, so should be OK to not reset
socialProvider: undefined, // in email/social we don't have multi-account yet, so should be OK to not reset
socialWindow: undefined, // in email/social we don't have multi-account yet, so should be OK to not reset
farcasterUrl: undefined // in email/social we don't have multi-account yet, so should be OK to not reset
```

But another issue is that `syncAccount` is called for different purposes, the calls other than account change might need to reset all?

Reply:
Refactored the ,syncAccount` and removed the unnecessary `resetAccount` and `status` prop which is doing the same with the `isConnected` and `isDisconnected` , see: https://wagmi.sh/core/api/actions/getAccount#isconnecting-isreconnecting-isconnected-isdisconnected